### PR TITLE
Akka.Actor: make `ITimeProvider` injectable into consuming classes

### PR DESCRIPTION
**Pull request by Aaronontheweb**
## Changes

Removes dependencies on the static `DateTimeNowTimeProvider` and instead has everything use the `IScheduler` instead, which provides the exact same functionality with identical implementations. This makes it possible to inject the `ITimeProvider` using the `ActorSystem`'s configuration instead.

This should enable us to be able to support [.NET 8's `TimeProvider`](https://learn.microsoft.com/en-us/dotnet/api/system.timeprovider) more easily in the future for testing purposes.

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Changes in public API reviewed, if any.

### CODE REVIEW

File: src/core/Akka/Actor/Scheduler/DateTimeNowTimeProvider.cs [hunk 1]
```diff
@@ -10,29 +10,19 @@
...
-        /// TBD
-        /// </summary>
+        
         public TimeSpan HighResMonotonicClock{get { return Util.MonotonicClock.ElapsedHighRes; }}
-
-        /// <summary>
-        /// TBD
-        /// </summary>
+        
         public static DateTimeOffsetNowTimeProvider Instance { get; } = new();
```
Comment (Aaronontheweb) [inline src/core/Akka/Actor/Scheduler/DateTimeNowTimeProvider.cs:26]: Basically, I removed all references to this type (all of which were contained inside Akka.Delivery and Akka.Cluster.Sharding.Delivery) and preferred to use the `IScheduler`'s implementation of these same methods instead, which are identical on the `HashedWheelTimer` base class.

File: src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveCore.DotNet.verified.txt [hunk 2]
```diff
@@ -589,6 +589,7 @@ namespace Akka.Actor
         public CoordinatedShutdownExtension() { }
         public override Akka.Actor.CoordinatedShutdown CreateExtension(Akka.Actor.ExtendedActorSystem system) { }
     }
+    [System.ObsoleteAttribute("This class will be removed in Akka.NET v1.6.0 - use the IScheduler instead.")]
```
Comment (Aaronontheweb) [inline src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveCore.DotNet.verified.txt:592]: Marked the `DateTimeNowTimeProvider` as `Obsolete` and slated for removal in v1.6.

**Review by Aaronontheweb — COMMENTED** [review]
Detailed my changes

**Review by Arkatufus — APPROVED** [review]
LGTM

**Pull request merged**